### ISSUE TITLE
How does liftbridge server use tls ?

### ISSUE THREAD

**Issue by linyuan0** [liftbridge-io/liftbridge#227]
I set up tls verification in nats-server, I want to connect to nats through liftbridge server, but it keeps reporting errors;
**nats-server config**
tls {
    cert_file: "server-cert.pem"
    key_file:  "server-key.pem"
    ca_file:   "rootCA.pem"
    verify: true
    timeout: 2
}

**liftbridge.yaml**
tls:
  cert: ./client-cert.pem
  key: ./client-key.pem
  client.auth.ca: ./rootCA.pem
  client.auth.enabled: true

**nats-server keeps reporting errors TLS handshake error: tls: client didn't provide a certificate**
**I want to know if there is an error in my configuration file or the liftbridge server cannot connect to nats-server via tls**

**Comment by tylertreat** [comment]
The tls config in the liftbridge configuration is for configuring TLS on the connection to Liftbridge, not for the connection from Liftbridge to NATS. Currently, TLS between Liftbridge and NATS is not implemented. I have added an issue to get this implemented (#228). Closing this as that work can be tracked in that issue.

**Comment by linyuan0** [comment]
@tylertreat  Thanks you very much.

**Comment by linyuan0** [comment]
@tylertreat 
I want to ask TLS between liftbridges again.
**liftbridge.yaml**
tls:
   cert: ./server-cert.pem
   key: ./server-key.pem
   #client.auth.ca: ./rootCA.pem
   #client.auth.enabled: true
**lift_sub.go**
        addrs := []string{"localhost:9292"}
        client, err := lift.Connect(addrs,lift.TLSCert("./server-cert.pem"))

I commented out #client.auth.enabled: true, the client can connect to liftbridge normally.
But I set client.auth.enabled: true, client return err.
**panic: rpc error: code = Unavailable desc = write tcp 10.33.36.108:44008->10.33.36.108:9292: write: broken pipe**
I checked the liftbridge example code tls.yaml did not set client.auth.enabled, when I set this property, what changes do I need to make for the client connection?

**Comment by tylertreat** [comment]
To do mutual TLS (`tls.client.auth.enabled: true`), you will need to create a CA cert (specified with `tls.client.auth.ca` config) and a client key and certificate that is signed with the CA cert. The example below calls these `ca-cert.pem`, `client-key.pem`, and `client-cert.pem` respectively.

```yaml
tls:
    key: server-key.pem
    cert: server-cert.pem
    client.auth.enabled: true
    client.auth.ca: ca-cert.pem
```

```go
certPool := x509.NewCertPool()
ca, err := ioutil.ReadFile("ca-cert.pem")
if err != nil {
	panic(err)
}
certPool.AppendCertsFromPEM(ca)
certificate, err := tls.LoadX509KeyPair("client-cert.pem", "client-key.pem")
if err != nil {
	panic(err)
}
config := &tls.Config{
	ServerName:   "localhost",
	Certificates: []tls.Certificate{certificate},
	RootCAs:      certPool,
}
client, err := lift.Connect([]string{"localhost:9292"}, lift.TLSConfig(config))
```

**Comment by tylertreat** [comment]
FYI TLS support between Liftbridge and NATS was merged in #229 and will be part of the forthcoming v1.2.0 release.

**Comment by linyuan0** [comment]
@tylertreat Thanks!
